library/Notifications/Daemon/Daemon.php

```diff
@@ -0,0 +1,361 @@
+            ++$deletions;
+        }
+
+        if ($deletions > 0) {
+            self::$logger::info(self::PREFIX . "housekeeping cleaned " . $deletions . " stale browser sessions");
+        }
+
+        self::$logger::debug(self::PREFIX . "finished housekeeping job");
+    }
+
+    /**
+     * Process new notifications (if there are any)
+     *
+     * @return void
+     */
+    protected function processNotifications(): void
+    {
+        $numOfNotifications = 0;
+
+        if ($this->lastIncidentId === null) {
+            // get the newest incident identifier
+            /** @var IncidentHistory $latestIncidentNotification */
+            $latestIncidentNotification = IncidentHistory::on(Database::get())
+                ->filter(Filter::equal('type', 'notified'))
+                ->orderBy('id', 'DESC')
+                ->first();
+            if (! $latestIncidentNotification) {
+                // early return as we don't need to check for new entries if we don't have any at all
+                return;
+            }
+
+            $this->lastIncidentId = $latestIncidentNotification->id;
+            self::$logger::debug(
+                self::PREFIX
+                . "fetched latest incident notification identifier: <id: "
+                . $this->lastIncidentId
+                . ">"
+            );
+        }
+
+        // grab new notifications and the current connections
+        $notifications = IncidentHistory::on(Database::get())
+            ->filter(Filter::greaterThan('id', $this->lastIncidentId))
+            ->filter(Filter::equal('type', 'notified'))
+            ->orderBy('id', 'ASC')
+            ->with(['incident', 'incident.object']);
+        /** @var array<int, array<Connection>> $connections */
+        $connections = $this->server->getMatchedConnections();
+
+        /** @var IncidentHistory $notification */
+        foreach ($notifications as $notification) {
+            if (isset($connections[$notification->contact_id])) {
+                /** @var Incident $incident */
+                $incident = $notification->incident;
+
+                $tags = null;
+                /** @var ObjectIdTag $tag */
+                foreach ($incident->object->object_id_tag as $tag) {
+                    $tags[] = $tag;
+                }
+
+                if ($tags !== null) {
+                    $host = $service = '';
+
+                    foreach ($tags as $tag) {
+                        switch ($tag->tag) {
+                            case 'host':
+                                $host = $tag->value;
+
+                                break;
+                            case 'service':
+                                $service = $tag->value;
+
+                                break;
+                        }
+                    }
+
+                    self::$logger::warning(self::PREFIX . "Host: " . $host . " | Service: " . $service);
```

Comment:
Otherwise the log message for host would look something like this:
```
[daemon] - Host: test-host | Service:
```

Expected:
```
[daemon] - Host: dummy-902
```